### ISSUE TITLE
[BUG] Incomplete logs when "docker compose logs > filename &" is used and services are started one by one

### ISSUE THREAD

**Issue by ab-public** [docker/compose#10480]
### Description

When I redirect logs via `docker compose logs -f > filename &` to a file and start several services **one after the other** (this is needed in my setup), only the messages of the first two services appear in the file. The messages of the further started services are missing and the log file is incomplete. However, if I run "docker compose logs" directly (without redirecting), the messages are complete. See example...

Hint: With docker-compose 1.29.x this works as expected.

### Steps To Reproduce

Example for reproducing

**docker-compose.yml**

```
version: "3"
services:
  service-1:
    image: alpine
    hostname: service-1
    command: sh -c "while true; do echo hello from $(hostname); sleep 3; done"
  service-2:
    image: alpine
    hostname: service-2
    command: sh -c "while true; do echo hello from $(hostname); sleep 3; done"
  service-3:
    image: alpine
    hostname: service-3
    command: sh -c "while true; do echo hello from $(hostname); sleep 3; done"
  service-4:
    image: alpine
    hostname: service-4
    command: sh -c "while true; do echo hello from $(hostname); sleep 3; done"
```
**test.bash**

```
#!/usr/bin/env bash

docker compose up -d service-1

docker compose logs -t -f > example.log &
## start log capturing...

docker compose up -d service-2
## this service appears in file log

docker compose up -d service-3
## The messages of this and further started services do not appear in the log file
## But if you run "docker compose logs" they appear... 

docker compose up -d service-4
## ...

tail -f example.log
```

**My output**

The messages of example-service-3 and example-service-4 are missing in example.log !

```
❯ ./test.bash         
[+] Running 2/2
 ✔ Network example_default        Created                                                                         0.1s 
 ✔ Container example-service-1-1  Started                                                                         0.2s 
[+] Running 1/1
 ✔ Container example-service-2-1  Started                                                                         0.3s 
[+] Running 1/1
 ✔ Container example-service-3-1  Started                                                                         0.3s 
[+] Running 1/1
 ✔ Container example-service-4-1  Started                                                                         0.3s 
example-service-1-1  | 2023-04-19T11:47:13.601278918Z hello from service-1
example-service-1-1  | 2023-04-19T11:47:16.603528205Z hello from service-1
example-service-2-1  | 2023-04-19T11:47:17.003981961Z hello from service-2
example-service-1-1  | 2023-04-19T11:47:19.606058022Z hello from service-1
example-service-2-1  | 2023-04-19T11:47:20.006779015Z hello from service-2
example-service-1-1  | 2023-04-19T11:47:22.609133835Z hello from service-1
example-service-2-1  | 2023-04-19T11:47:23.010339727Z hello from service-2
^Ccanceled

❯ docker compose logs 
example-service-1-1  | hello from service-1
example-service-3-1  | hello from service-3
example-service-1-1  | hello from service-1
example-service-3-1  | hello from service-3
example-service-3-1  | hello from service-3
example-service-3-1  | hello from service-3
example-service-1-1  | hello from service-1
example-service-1-1  | hello from service-1
example-service-2-1  | hello from service-2
example-service-4-1  | hello from service-4
example-service-2-1  | hello from service-2
example-service-2-1  | hello from service-2
example-service-2-1  | hello from service-2
example-service-4-1  | hello from service-4
example-service-4-1  | hello from service-4
example-service-4-1  | hello from service-4

❯ docker compose down 
[+] Running 5/5
 ✔ Container example-service-1-1  Removed                                                                        10.6s 
 ✔ Container example-service-3-1  Removed                                                                        10.5s 
 ✔ Container example-service-2-1  Removed                                                                        10.4s 
 ✔ Container example-service-4-1  Removed                                                                        10.7s 
 ✔ Network example_default        Removed                                                                         0.5s 
```

### Compose Version

```Text
❯ docker compose version
Docker Compose version v2.17.2
```
```


### Docker Environment

```Text
❯ docker version
Client: Docker Engine - Community
 Version:           23.0.3
 API version:       1.42
 Go version:        go1.19.7
 Git commit:        3e7cbfd
 Built:             Tue Apr  4 22:05:48 2023
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Engine - Community
 Engine:
  Version:          23.0.3
  API version:      1.42 (minimum version 1.12)
  Go version:       go1.19.7
  Git commit:       59118bf
  Built:            Tue Apr  4 22:05:48 2023
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.20
  GitCommit:        2806fc1057397dbaeefbea0e4e17bddfbd388f38
 runc:
  Version:          1.1.5
  GitCommit:        v1.1.5-0-gf19387a
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```
```


### Anything else?

_No response_

**Comment by ndeloof** [comment]
I tried your example compose file without redirection, and I get logs from second service attached when started _after_ `compose logs` command, but not for service 3 nor 4 - this is reproducible

investigating

**Comment by ndeloof** [comment]
https://github.com/docker/compose/pull/10493 has been merged but is not included in 2.17.3 release
Maybe you can build compose from sources (run `make install`) and test it to confirm issue is fixed
